test/integration/policy_set_test.go

```diff
@@ -0,0 +1,247 @@
+				1,
+			).Should(BeNil())
+
+			templates, found, err := unstructured.NestedSlice(rootPolicy.Object, "spec", "policy-templates")
+			Expect(err).Should(BeNil())
+			Expect(found).Should(BeTrue())
+			Expect(len(templates)).Should(Equal(1))
+
+			By("Patching placement rule " + testPolicyName + "-plr")
+			output, err = utils.KubectlWithOutput("patch", "-n", userNamespace, "placementrule.apps.open-cluster-management.io/"+testPolicySetName+"-plr",
+				"--type=json", "-p=[{\"op\": \"replace\", \"path\": \"/spec/clusterSelector/matchExpressions\", \"value\":[{\"key\": \"name\", \"operator\": \"In\", \"values\": ["+clusterNamespace+"]}]}]",
+				"--kubeconfig="+kubeconfigHub)
+			By("Patching placement rule result is " + output)
+			Expect(err).To(BeNil())
+
+			By("Checking " + testPolicyName + " on managed cluster in ns " + clusterNamespace)
+			policyRsrc := clientHubDynamic.Resource(testcommon.GvrPolicy)
+			Eventually(
+				func() error {
+					var err error
+					_, err = policyRsrc.Namespace(clusterNamespace).Get(
+						context.TODO(), userNamespace+"."+testPolicyName, metav1.GetOptions{},
+					)
+					return err
+				},
+				defaultTimeoutSeconds*2,
+				1,
+			).Should(BeNil())
+
+			opt := metav1.ListOptions{
+				LabelSelector: common.RootPolicyLabel + "=" + userNamespace + "." + testPolicyName,
+			}
+
+			By("Patching replicated policy statuses - NonCompliant")
+			replicatedPlcList := utils.ListWithTimeout(clientHubDynamic, testcommon.GvrPolicy, opt, 1, true,
+				defaultTimeoutSeconds)
+			for _, replicatedPlc := range replicatedPlcList.Items {
+				replicatedPlc.Object["status"] = &policiesv1.PolicyStatus{
+					ComplianceState: policiesv1.NonCompliant,
+					Status: []*policiesv1.CompliancePerClusterStatus{
+						{
+							ClusterName:      clusterNamespace,
+							ClusterNamespace: clusterNamespace,
+							ComplianceState:  policiesv1.NonCompliant,
+						},
+					},
+				}
+				_, err = clientHubDynamic.Resource(testcommon.GvrPolicy).Namespace(replicatedPlc.GetNamespace()).UpdateStatus(
+					context.TODO(), &replicatedPlc, metav1.UpdateOptions{},
+				)
+				Expect(err).To(BeNil())
+			}
+
+			By("Checking the status of policy set")
+			yamlPlc := utils.ParseYaml("../resources/policy_set/statuscheck-1.yaml")
+			// set checking yaml clusterName and clusterNamespace by env variable
+			By("Updating the checking YAML clusterName to " + clusterNamespace)
+			yamlPlc = testcommon.UpdateClusterName(yamlPlc, clusterNamespace)
+
+			policySetRsrc := clientHubDynamic.Resource(testcommon.GvrPolicySet)
+			var rootPlcSet *unstructured.Unstructured
+			Eventually(
+				func() error {
+					var err error
+					rootPlcSet, err = policySetRsrc.Namespace(userNamespace).Get(
+						context.TODO(), testPolicySetName, metav1.GetOptions{},
+					)
+					return err
+				},
+				defaultTimeoutSeconds*2,
+				1,
+			).Should(BeNil())
+
+			Expect(rootPlcSet.Object["status"]).Should(utils.SemanticEqual(yamlPlc.Object["status"]))
```

Comment:
Should use `Eventually` because you may have to wait until the status of policyset to match expected value.